lib/svg-sprite/shape.js

```diff
@@ -513,20 +511,60 @@ SVGShape.prototype._determineDimensions = function(cb) {
 	// If the viewBox attribute didn't suffice: Render the SVG image
 	if (!this.width || !this.height) {
 		var that				= this;
-	    execFile(phantomjs, [dimensionsPhantomScript, this.getSVG(false), 'file://' + this.source.path], function (err, stdout, stderr) {
-	        if (err) {
-	            cb(err);
-	        } else if (stdout.length > 0) { // PhantomJS always outputs to stdout.
-	        	var dimensions	= JSON.parse(stdout.toString().trim());
-	        	that.width		= that._round(dimensions.width);
-	        	that.height		= that._round(dimensions.height);
-	        	cb(null);
-	        } else if (stderr.length > 0) {
-	            cb(new Error(stderr.toString().trim()));
-	        } else {
-	            cb(new Error('PhantomJS didn\'t return dimensions for "' + that.name + '"'));
-	        }
-	    });
+		var _browser, _page, _nodeHandle;
+		try {
+			puppeteer.launch()
```

Comment:
Start a new browser instance is a very expensive operation. I think worth to implement a singleton for the browser instantiation  and dispose the the browser object when the `svg-sprite` finish the job.